src/lib/stable_structures/stable_b_tree_map.ts

```diff
@@ -1,24 +1,24 @@
 import { None, Opt, Some } from '../candid/types/constructed/opt';
 import { nat64 } from '../candid/types/primitive/nats/nat64';
 import { nat8 } from '../candid/types/primitive/nats/nat8';
-import { encode, decode } from '../candid/serde';
+import { decode } from '../candid/serde';
 
 export interface Serializable {
     toBytes: (data: any) => Uint8Array;
     fromBytes: (bytes: Uint8Array) => any;
 }
 
 export function StableBTreeMap<Key = any, Value = any>(
```

Comment:
Now that this function can throw, I think it's important to document that as a JS doc comment. I'd use the [`@throw`](https://jsdoc.app/tags-throws) tag. Now would also be a good time to add more documentation explaining how to use StableBTreeMap, for example maybe adding `@typeparam` tags to explain the `Key`, and `Value` generics.

Reply:
Documentation is good but nothing really throws here, isSerializable is just a type assertion it's not really meant to do anything at runtime, though it will.

This is a pattern that comes up often, and generally I think we don't want to rely on runtime type checking. The throw is just for the compiler.